test/unit/services/policies/policies_list_service_test.rb

```diff
@@ -1,18 +1,126 @@
+# frozen_string_literal: true
+
 require 'test_helper'
 
 class Policies::PoliciesListServiceTest < ActiveSupport::TestCase
 
   def setup
     @service = Policies::PoliciesListService
+    ThreeScale.config.sandbox_proxy.stubs(apicast_registry_url: 'https://apicast-staging.proda.example.com/policies')
   end
 
-  def test_call
-    ThreeScale.config.sandbox_proxy.stubs(apicast_registry_url: 'https://apicast-staging.proda.example.com/policies')
+  GATEWAY_API_MANAGEMENT_RESPONSE = {
+    policies: {
+      cors: [{
+        version: 'builtin',
+        description: 'CORS Policy'
+      }],
+      tls: [{
+        version: 'builtin',
+        description: 'TLS Policy'
+      }]
+    }
+  }.freeze
+
+  test 'call with error response on gateway side' do
+    stub_request(:get, "https://apicast-staging.proda.example.com/policies")
+      .to_return(status: 502, body: '{"error":"A server error occured"}',
+                 headers: { 'Content-Type' => 'application/json' })
+
+    account = FactoryBot.build_stubbed(:simple_provider)
+    assert_nil @service.call(account)
+  end
+
+  test 'call with no access to registry' do
     stub_request(:get, "https://apicast-staging.proda.example.com/policies")
-      .to_return(status: 200, body: "{\"policies\":[{\"schema\":\"1\"}]}",
-                  headers: { 'Content-Type' => 'application/json' })
+      .to_return(status: 200, body: GATEWAY_API_MANAGEMENT_RESPONSE.to_json,
+                 headers: { 'Content-Type' => 'application/json' })
+
+    account = FactoryBot.build_stubbed(:simple_provider)
+    account.expects(:provider_can_use?).with(:policy_registry).returns(false)
+    account.expects(:policies).never
+
+    policies = @service.call(account)
+    assert_equal GATEWAY_API_MANAGEMENT_RESPONSE[:policies].as_json, policies
+  end
+
+  test 'call with custom policies' do
+    stub_request(:get, "https://apicast-staging.proda.example.com/policies")
+      .to_return(status: 200, body: GATEWAY_API_MANAGEMENT_RESPONSE.to_json,
+                 headers: { 'Content-Type' => 'application/json' })
+
+    account = FactoryBot.build_stubbed(:simple_provider)
+
+    cors_v2 = {
+      version: '2.0.0',
+      description: 'CORS Policy V2'
+    }
+
+    header_v1 = {
+      version: '1.1.5',
+      description: 'Header Policy V1'
+    }
+
+    policies = [
+      Policy.new(name: 'cors', version: '2.0.0', schema: cors_v2),
```

Comment:
We have a factory for policy in case you prefer.

Reply:
I think we do not need factory for simple tests like this.
I only need unpersisted Policy and with factory it would be `FactoryBot.build_stubbed(:policy, ... attributes .. )` which is longer and unnecessary